clients/src/test/java/org/apache/kafka/clients/MetadataTest.java

```diff
@@ -1071,18 +1071,25 @@ protected boolean retainTopic(String topic, boolean isInternal, long nowMs) {
     public void testUpdatePartitionLeadership() {
         Time time = new MockTime();
```

Comment:
We probably need to refactor this test a certain point.  I tried to do it but as I moved something, we immediately exceeded the 160 line limit.

Reply:
sounds good - i was trying to limit the scope of change that's why i left this comment. i'll refactor this.

Reply:
i did file this jira: https://issues.apache.org/jira/browse/KAFKA-16027

Reply:
Sorry for confusion, I think you were right to limit the scope of the change. Let's do the refactor as a followup.

Reply:
Alright - I'll revert some commits and re-push